liberty/lib/Planner/Planner.cpp

```diff
@@ -266,6 +279,69 @@ namespace liberty {
     assert(pdg != nullptr && "PDG is null?");
     writeGraph<PDG>(pdgName, pdg);
 
+    if(SlampCheck) {
+      uint32_t edgeCount = 0;
+      uint32_t diffCount = 0;
+      auto remed_slamp = &getAnalysis<SLAMPLoadProfile>(); 
+      auto remed_slamp_aa = std::make_unique<SlampOracleAA>(remed_slamp); 
+      auto remed_lamp = &getAnalysis<LAMPLoadProfile>(); 
+      auto remed_lamp_aa = std::make_unique<LampOracle>(remed_lamp); 
+
+      for(auto &edge : make_range(pdg->begin_edges(), pdg->end_edges())) {
+        if(!pdg->isInternal(edge->getIncomingT()) || !pdg->isInternal(edge->getOutgoingT()))
+          continue;
+
+        Instruction* src = dyn_cast<Instruction>(edge->getOutgoingT());
+        Instruction* dst = dyn_cast<Instruction>(edge->getIncomingT());
+        assert(src && dst && "src/dst not instructions in PDG?");
+        bool loopCarried = false;
+        edgeCount++;
+        DataDepType dataDepTy;
+        if(dyn_cast<CallBase>(src) || dyn_cast<CallBase>(dst))
+          continue;
+        if(edge->isRAWDependence())
```

Comment:
The logic here is weird. Maybe do the check `if(!edge->isRAWDependence()) continue` and hardcode RAW in the `memdep` queries.